client/blocks/subscription-length-picker/index.jsx

```diff
@@ -0,0 +1,193 @@
+import { getSelectedSiteId } from '../../state/ui/selectors';
+import { abtest } from 'lib/abtest';
+import { getPlan, applyTestFiltersToPlansList } from 'lib/plans';
+import { TERM_MONTHLY } from 'lib/plans/constants';
+import SubscriptionLengthOption from './option';
+
+export class SubscriptionLengthPicker extends React.Component {
+	static propTypes = {
+		initialValue: PropTypes.string,
+		currencyCode: PropTypes.string.isRequired,
+		productsWithPrices: PropTypes.array.isRequired,
+		onChange: PropTypes.func.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onChange: () => null,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			checked: props.initialValue,
+		};
+	}
+
+	render() {
+		const { productsWithPrices, translate } = this.props;
+		return (
+			<div className="subscription-length-picker">
+				<div className="subscription-length-picker__header">
+					{ translate( 'Choose the length of your subscription', {
+						comment:
+							'We offer a way to change billing term from default bill every 1 year to something ' +
+							'else like bill once every 2 years - this is related header.',
+					} ) }
+				</div>
+
+				<div className="subscription-length-picker__options">
+					{ productsWithPrices.map( p => this.renderProduct( p ) ) }
+				</div>
+			</div>
+		);
+	}
+
+	renderProduct = p => {
+		const { plan, planSlug } = p;
+
+		return (
+			<div className="subscription-length-picker__option-container" key={ planSlug }>
+				<SubscriptionLengthOption
+					term={ plan.term }
+					checked={ planSlug === this.state.checked }
+					price={ myFormatCurrency( p.priceFull, this.props.currencyCode ) }
+					pricePerMonth={ myFormatCurrency( p.priceMonthly, this.props.currencyCode ) }
+					savePercent={ Math.round( 100 * ( 1 - p.priceMonthly / this.getHighestMonthlyPrice() ) ) }
+					value={ planSlug }
+					onCheck={ this.handleCheck }
+				/>
+			</div>
+		);
+	};
+
+	getHighestMonthlyPrice() {
+		return Math.max( ...this.props.productsWithPrices.map( p => Number( p.priceMonthly ) ) );
+	}
+
+	handleCheck = ( { value } ) => {
+		this.setState( {
+			checked: value,
+		} );
+		this.props.onChange( { value } );
+	};
+}
+
+class SubscriptionLengthPickerContainer extends React.Component {
+	static defaultProps = {
+		...SubscriptionLengthPicker.defaultProps,
+
+		fetchingPlans: PropTypes.bool.isRequired,
+		fetchingProducts: PropTypes.bool.isRequired,
+		requestProductsList: PropTypes.func.isRequired,
+		requestPlans: PropTypes.func.isRequired,
+	};
+
+	componentWillMount() {
+		if ( this.props.productsWithPrices.length === 0 ) {
+			if ( ! this.props.fetchingProducts ) {
+				this.props.requestProductsList();
+			}
+			if ( ! this.props.fetchingPlans ) {
+				this.props.requestPlans();
+			}
+		}
+	}
+
+	render() {
+		return <SubscriptionLengthPicker { ...this.props } />;
+	}
+}
+
+const EPSILON = 0.009;
+
+function myFormatCurrency( price, code ) {
```

Comment:
Please use https://github.com/Automattic/wp-calypso/tree/master/client/lib/format-currency If we need to format money on the client. There are many corner cases here.

Reply:
Note that we may also figure out if a currency has cents by the precision value in `client/lib/format-currency/currencies.js` more consistently. Anything with a precision value > 0 has "cents"

Reply:
You're also welcome to add a new util function that does this in the library.

Reply:
This here determines if specific amount (rather than the currency) has cents or not, basically we want to display "14" instead of "14.00" but keep "14.26" as it is - I may add such utility function in another PR if I see this use case in other places, thanks!